src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/Startup.cs

```diff
@@ -0,0 +1,315 @@
+#else
+using _MicroserviceTemplate_.EF.PostgreSql.Services;
+#endif
+
+namespace _MicroserviceTemplate_.WebApi
+{
+    /// <summary>
+    /// Startup.
+    /// </summary>
+    public class Startup
+    {
+        /// <summary>
+        /// Configuration.
+        /// </summary>
+        public IConfiguration Configuration { get; }
+
+        /// <summary>
+        /// Creates startup instance.
+        /// </summary>
+        /// <param name="configuration">Configuration.</param>
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        /// <summary>
+        /// Configures application services.
+        /// </summary>
+        /// <param name="services">Services.</param>
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc();
+            services.AddCors();
+
+            #if SQLSERVER
+            ConfigureSqlServerContext(services);
+            #else
+            ConfigurePostgreSqlContext(services);
+            #endif
+
+            services.AddAutoMapper(typeof(_MicroserviceTemplate_AutoMapperProfile));
+
+            services.AddHttpClient();
+
+            services
+                .AddControllers()
+                .AddApplicationPart(Assembly.GetAssembly(typeof(ITController)));
+
+            AddVersioning(services);
+
+            services.AddSwagger(Configuration);
+
+            AddServices(services);
+            AddAuthorization(services);
+        }
+
+        /// <summary>
+        /// Configures the application.
+        /// </summary>
+        /// <param name="app">Application builder.</param>
+        /// <param name="env">Environment.</param>
+        /// <param name="apiVersionDescriptionProvider">API version description provider.</param>
+        public void Configure(
+            IApplicationBuilder app,
+            IWebHostEnvironment env,
+            IApiVersionDescriptionProvider apiVersionDescriptionProvider)
+        {
+            app.UseCors(options =>
+            {
+                options
+                    .WithOrigins(Configuration.GetValue<string>("AllowedHosts"))
+                    .AllowAnyHeader()
+                    .AllowAnyMethod();
+            });
+
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+            else
+            {
+                app.UseHsts();
+                app.UseHttpsRedirection();
+            }
+
+            UseDatabaseMigrations(app);
+
+            app.UseRouting();
+            app.UseAuthentication();
+            app.UseAuthorization();
+
+            app.UseStaticFiles();
+
+            app.UseSwagger(apiVersionDescriptionProvider, Configuration);
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapControllers();
+            });
+        }
+
+        private void ConfigureSqlServerContext(IServiceCollection services)
```

Comment:
@fuadagazade i do not like this duplication , you can use same funtion and the do the ifdef only for the line with the 
  services.AddDbContext

Reply:
Changed